pkg/auth/asynccache.go

```diff
@@ -27,14 +32,18 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
 		cachingFunc:  cachingFunc,
 	}
 
-	item, err := cachingFunc(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to setup an async cache - caching func returned error: %w", err)
+	var err error
+	for retries := 0; retries < initializationRetries; retries++ {
+		item, err := cachingFunc(ctx)
+		if err == nil {
+			c.cachedItem = item
+			return c, nil
+		}
+		klog.V(4).Infof("retrying async cache setup - attempt %v of %v", retries, initializationRetries)
+		time.Sleep(initializationRetryDelay)
```

Comment:
```suggestion
		select{
		case <- ctx.Done():
			return nil, ctx.Error()
		case <-time.After(initializationRetryDelay):
		}
```

perhaps this, so that cancellation is honored?

Reply:
replace the logic with `k8s.io/apimachinery/pkg/util/wait.UntilWithContext()`

Reply:
@deads2k @stlaz @jhadvig I've addressed this feedback.